Index: b/make/autoconf/autogen.sh
===================================================================
--- a/make/autoconf/autogen.sh
+++ b/make/autoconf/autogen.sh
@@ -52,7 +52,9 @@ else
   TIMESTAMP=`date +%s`
 fi
 
+if test "x${AUTOCONF}" = x; then
 AUTOCONF="`which autoconf 2> /dev/null | grep -v '^no autoconf in'`"
+fi
 
 if test "x${AUTOCONF}" = x; then
   echo "You need autoconf installed to be able to regenerate the configure script"
