Index: b/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java
===================================================================
--- a/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java
+++ b/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java
@@ -127,7 +127,7 @@ public class WorkArounds {
         }
 
         if (!msgOptionSeen) {
-            doclintOpts.add(DocLint.XMSGS_OPTION);
+            return;
         }
 
         String sep = "";
